src/bin/tikv-server.rs

```diff
@@ -248,8 +249,27 @@ fn run_raft_server(pd_client: RpcClient, cfg: &TiKvConfig, security_mgr: Arc<Sec
     server
         .start(server_cfg, security_mgr)
         .unwrap_or_else(|e| fatal!("failed to start server: {:?}", e));
+
+    let server_cfg = cfg.server.clone();
+    let mut status_enabled = !server_cfg.status_addr.is_empty();
+
+    // Create an HTTP server.
+    let mut http_server = HttpServer::new(server_cfg.status_thread_pool_size);
+    if status_enabled {
+        // Start the HTTP server.
+        if let Err(e) = http_server.start(server_cfg.status_addr) {
+            error!("failed to bind addr, error: {:?}", e);
+            status_enabled = false;
+        }
+    }
+
     signal_handler::handle_signal(Some(engines));
 
+    if status_enabled {
```

Comment:
Better stop it after server is completely shutdown, so that service like health check can rely on this.